compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1609,6 +1609,7 @@ public void visit(BLangTupleVariable varNode, AnalyzerData data) {
         typeChecker.checkExpr(varNode.expr, currentEnv, varNode.getBType(), data.prevEnvs,
                 data.commonAnalyzerData);
         checkSelfReferencesInVarNode(varNode, varNode.expr, data);
+        analyzeNode(varNode.typeNode, data);
```

Comment:
By doing this at the end we don't get the error where we return early.

E.g.,

```ballerina
public function main() {
    [@UndefinedAnnotation int, int] [a, b, c] = [1, 2, 3];
}
```

Reply:
Fixed